program/src/processor.rs

```diff
@@ -763,6 +813,15 @@ impl Processor {
         let clock = &Clock::from_account_info(clock_info)?;
         let option_lockup_authority_info = next_account_info(account_info_iter).ok();
 
+        // JC: I hope I'm missing something, but is this instruction currently broken
+        // if there's a lockup in force and you want to change the withdraw
+        // authority? It looks like you're fixing it here!
+        // In the old implementation, I don't think it'll ever add the lockup
+        // authority to the signers hashset.
+        // But to make it consistent with `authorize_checked_with_seed`, how about
+        // passing `true` for checked? I don't think it ends up making a difference,
+        // since if the lockup is not in force anymore, you can just not pass
+        // the custodian at all, and it'll work.
```

Comment:
i believe youre right... it looks like the existing stake program inits the signers object with all signers. authorize uses this object, but authorize with need ignores it and constructs a new one, and only adds the seed account, so it will fail if theres a lockup. i dont recall noticing it was wrong, i think i just did the right thing automatically once essential validation code was no longer split between two files